jupyterhub_home_nfs/generate.py

```diff
@@ -181,8 +193,12 @@ def reconcile_projfiles(self, *, is_dirty=False):
         # Sort to provide consistent ordering across runs
         homedirs = []
         for path in self.paths:
-            # Create the directory if it doesn't exist
-            os.makedirs(path, exist_ok=True)
+            # Create the directory if it doesn't exist and make sure is owned by uid:gid
+            try:
+                os.makedirs(path)
+                os.chown(path, self.uid, self.gid)
```

Comment:
This should be then unconditional and throw errors if it doesn't work, and crash with an exception. our monitoring will then catch that.